src/index.js

```diff
@@ -360,6 +361,72 @@ class Compute extends common.Service {
       }
     );
   }
+  /**
+   * Create an instance template.
+   *
+   * @see [Instance Templates: insert API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceTemplates/insert}
+   *
+   * @param {string} name - The name of the target image.
+   * @param {object} parameters - The settings of the template, see the
+   *     [Instance Templates: insert API documentation](https://cloud.google.com/compute/docs/reference/v1/instanceTemplates/insert).
+   * @param {object} [options] - See the
+   *     [Instance Templates: insert API documentation](https://cloud.google.com/compute/docs/reference/v1/instanceTemplates/insert).
+   * @param {function} callback - The callback function.
+   * @param {?error} callback.err - An error returned while making this request.
+   * @param {Operation} callback.operation - An operation object that can be used
+   *     to check the status of the request.
+   * @param {object} callback.apiResponse - The full API response.
+   *
+   * @example
+   * const Compute = require('@google-cloud/compute');
+   * const compute = new Compute();
+   *
+   * const options = {
+   *
+   * }
+   * compute.createInstanceTemplate('new-template', options, function(err, operation, apiResponse) {
+   *   // `operation` is an Operation object that can be used to check the status
+   *   // of instance template creation.
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * compute.createInstanceTemplate('new-image', options).then(function(data) {
+   *   var operation = data[0];
+   *   var apiResponse = data[1];
+   * });
+   */
+  createInstanceTemplate(name, options, callback) {
```

Comment:
This method needs tests.

Reply:
Does it? I'm mostly interested in testing new InstanceTemplate which is tested in test/instance-template.js

Reply:
Yes, we would want something like:

- should make the correct request
  - stubs "this.request"
  - confirms the method is "POST"
  - confirms the URI is "/global/instanceTemplates"
  - confirms the JSON body is `{name: name}`
- should send options in request body
  - stubs "this.request" and confirms the user-provided options are part of the request body
- errors
  - should return the error and API response
    - mocks "this.request" and returns an error and API response
    - confirms the error and API response argument returned are what the caller received
- success
  - should return an instance template
    - mocks "this.instanceTemplate", confirming the "name" received is what the user provided
    - mocks "this.operation", confirming an operation was created from the API response object's "name" property
    - confirms `metadata` was assigned with the API response
    - confirms the user receives the response arguments in order correctly to the callback